core/src/main/java/io/smallrye/openapi/runtime/io/info/InfoIO.java

```diff
@@ -1,82 +1,40 @@
 package io.smallrye.openapi.runtime.io.info;
 
-import java.util.Optional;
-
 import org.eclipse.microprofile.openapi.models.info.Info;
 import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.AnnotationValue;
 
-import io.smallrye.openapi.api.models.info.InfoImpl;
 import io.smallrye.openapi.runtime.io.IOContext;
-import io.smallrye.openapi.runtime.io.IOContext.OpenApiVersion;
-import io.smallrye.openapi.runtime.io.IoLogging;
 import io.smallrye.openapi.runtime.io.ModelIO;
 import io.smallrye.openapi.runtime.io.Names;
 
 public class InfoIO<V, A extends V, O extends V, AB, OB> extends ModelIO<Info, V, A, O, AB, OB> {
 
-    private static final String PROP_TERMS_OF_SERVICE = "termsOfService";
-    private static final String PROP_TITLE = "title";
-    private static final String PROP_VERSION = "version";
-    private static final String PROP_DESCRIPTION = "description";
     private static final String PROP_LICENSE = "license";
     private static final String PROP_CONTACT = "contact";
-    private static final String PROP_SUMMARY = "summary";
 
     public InfoIO(IOContext<V, A, O, AB, OB> context) {
         super(context, Names.INFO, Names.create(Info.class));
     }
 
     @Override
-    public Info read(AnnotationInstance annotation) {
-        IoLogging.logger.annotation("@Info");
-        Info info = new InfoImpl();
-        info.setTitle(value(annotation, PROP_TITLE));
-        info.setDescription(value(annotation, PROP_DESCRIPTION));
-        info.setSummary(value(annotation, PROP_SUMMARY));
-        info.setTermsOfService(value(annotation, PROP_TERMS_OF_SERVICE));
-        info.setContact(contactIO().read(annotation.value(PROP_CONTACT)));
-        info.setLicense(licenseIO().read(annotation.value(PROP_LICENSE)));
-        info.setVersion(value(annotation, PROP_VERSION));
-        info.setExtensions(extensionIO().readExtensible(annotation));
-        return info;
-    }
-
-    /**
-     * Reads an {@link Info} OpenAPI node.
-     *
-     * @param node the json node
-     * @return Info model
-     */
-    @Override
-    public Info readObject(O node) {
-        IoLogging.logger.singleJsonNode("Info");
```

Comment:
With the removal of this `readObject`, the call will now go to `ModelIO.readObject` which will throw an `UnsupportedOperationException`.

It looks like only `OpenAPIDefinitionIO` still has `readObject` implemented which delegates to `ModelIO.readObject(Class, O)`. I assume that one was kept to keep backwards compatibility?

OpenLiberty was previously using `InfoIO.readObject` to parse an `Info` object for our configuration. I can change that to `InfoIO.readObject(Info.class, node);` but I am wondering if that's a reasonable way to parse an arbitrary model object.

Reply:
I think that is a reasonable way to parse it. The only thing remaining with special/overridden JSON handling is the `SchemaIO`. 

We'll need to talk through an approach for how a parsing API should look more generally so there can be a stable contract that is exposed for it. I think it's close, with `ModelIO.readObject(Class, O)` basically having the needed functionality, but there is still some further refactoring to do I think.